syndicate/utils.py

```diff
@@ -0,0 +1,247 @@
+    @see https://pygithub.readthedocs.io/en/latest/github_objects/File.html#github.File.File
+    """
+    assert os.getenv("GITHUB_SHA"), "missing GITHUB_SHA"
+    # NOTE
+    # Explicitly using GITHUB_SHA to ensure we always have access to the changed
+    # files even if other steps generate commits.
+    return repo().get_commit(os.getenv("GITHUB_SHA")).files
+
+def file_contents(filepath):
+    """
+    Returns a `ContentFile` object of the matching the given path in latest known
+    commit to this repo.
+    @see https://pygithub.readthedocs.io/en/latest/github_objects/ContentFile.html#github.ContentFile.ContentFile
+    @see :func:`~syndicate.utils.parent_sha`
+    """
+    # NOTE
+    # Using the latest known commit to ensure we capture any modifications made
+    # to the post frontmatter by previous actions.
+    return repo().get_contents(filepath, ref=parent_sha())
+
+def get_posts(post_dir=os.getenv('SYNDICATE_POST_DIR', 'posts')):
+    """
+    Returns the latest known :func:`~syndicate.utils.file_contents` of the files
+    added and modified in the commit that triggered this Github workflow.
+    """
+    files = get_trigger_payload()
+    assert files, "target commit was empty"
+
+    posts = [file for file in files if file.filename.startswith(post_dir)]
+    return [
+        file_contents(post.filename)
+        for post in posts
+        if post.status != 'deleted'  # ignore deleted files
+    ]
+
+def fronted(post):
+    """
+    Returns the :py:class:`frontmatter.Post` representation of the given
+    :func:`~syndicate.utils.file_contents` object.
+
+    If `post` is actually already a `frontmatter.Post`, this is a no-op.
+    """
+    assert post, "missing post"
+    if type(post) == frontmatter.Post:
+        return post
+    raw_contents = post.decoded_content.decode('utf-8')
+    return frontmatter.loads(raw_contents)
+
+def silo_key_for(silo):
+    """Returns a formatted string used to identify a silo ID in post frontmatter."""
+    return f'{silo.lower()}_silo_id'
+
+def silo_id_for(post, silo):
+    """
+    Retrieves the ID appropriate for `silo` from the frontmatter of the given
+    `post`; returns None if no relevant ID exists.
+    """
+    assert post, "missing post"
+    assert silo, "missing silo"
+    return fronted(post).get(silo_key_for(silo))
+
+def mark_syndicated_posts(silo_ids_by_path, fronted_posts_by_path):
+    """
+    Injects the given silo IDs for the given posts into their frontmatter
+    and commits the updated posts back to this repo.
+
+    If a silo ID already exists in a given post, that's fine: we assume IDs don't
+    change, and so we don't try to change them.
+
+    Returns a dictionary which is the response of the commit request.
+    """
+    assert silo_ids_by_path, "missing silo IDs"
+    assert fronted_posts_by_path, "missing fronted posts"
+
+    updated_fronted_posts_by_path = {}
+    silos_included = set()
+    for (path, silo_ids_by_silo) in silo_ids_by_path.items():
```

Comment:
parens are optional here, I'd use the following instead:
```python
for path, silo_ids_by_silo in silo_ids_by_path.items():
    ...
```

Reply:
Done in https://github.com/dabrady/syndicate/pull/1/commits/598380b3ecc84072f3dbb1eda6d9924a51d671c7